diff --git a/CMakeLists.txt b/CMakeLists.txt
index a105c0d..640546c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ project(bear
     LANGUAGES C CXX)
 
 enable_testing()
-find_package(GTest REQUIRED)
+include(gtest.cmake)
 find_package(Threads REQUIRED)
 
 add_subdirectory(source)
diff --git a/gtest.cmake b/gtest.cmake
new file mode 100644
index 0000000..c6b4aea
--- /dev/null
+++ b/gtest.cmake
@@ -0,0 +1,17 @@
+set(GOOGLETEST_ROOT gtest/googletest CACHE STRING "Google Test source root")
+
+include_directories(SYSTEM
+    ${PROJECT_SOURCE_DIR}/${GOOGLETEST_ROOT}
+    ${PROJECT_SOURCE_DIR}/${GOOGLETEST_ROOT}/include
+    )
+
+set(GOOGLETEST_SOURCES
+    ${PROJECT_SOURCE_DIR}/${GOOGLETEST_ROOT}/src/gtest-all.cc
+    ${PROJECT_SOURCE_DIR}/${GOOGLETEST_ROOT}/src/gtest_main.cc
+    )
+
+foreach(_source ${GOOGLETEST_SOURCES})
+    set_source_files_properties(${_source} PROPERTIES GENERATED 1)
+endforeach()
+
+add_library(gtest ${GOOGLETEST_SOURCES})
diff --git a/source/intercept_a/test/CMakeLists.txt b/source/intercept_a/test/CMakeLists.txt
index f6058a5..8ec4c53 100644
--- a/source/intercept_a/test/CMakeLists.txt
+++ b/source/intercept_a/test/CMakeLists.txt
@@ -5,6 +5,6 @@ add_executable(libintercept_unit_test
 )
 
 target_link_libraries(libintercept_unit_test intercept_a)
-target_link_libraries(libintercept_unit_test gtest gtest_main ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(libintercept_unit_test gtest ${CMAKE_THREAD_LIBS_INIT})
 
 add_test(NAME LibInterceptUnitTests COMMAND libintercept_unit_test)
diff --git a/source/libexec_a/test/CMakeLists.txt b/source/libexec_a/test/CMakeLists.txt
index cb84bd4..1a7a6cc 100644
--- a/source/libexec_a/test/CMakeLists.txt
+++ b/source/libexec_a/test/CMakeLists.txt
@@ -6,6 +6,6 @@ add_executable(libexec_unit_test
 )
 
 target_link_libraries(libexec_unit_test exec_a)
-target_link_libraries(libexec_unit_test gtest gtest_main ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(libexec_unit_test gtest ${CMAKE_THREAD_LIBS_INIT})
 
 add_test(NAME LibExecUnitTests COMMAND libexec_unit_test)
